packages/amplication-server/src/core/resource/resource.service.ts

```diff
@@ -1552,10 +1552,28 @@ export class ResourceService {
   }
 
   async resources(args: FindManyResourceArgs): Promise<Resource[]> {
+    const { serviceTemplateId, ...where } = args.where;
+    delete args.where?.serviceTemplateId;
+
+    let resourceIds: string[] = undefined;
+    if (serviceTemplateId) {
+      if (!args.where?.project?.id) {
+        throw new Error(
+          "project.id is required when searching by serviceTemplateId"
+        );
+      }
+
+      resourceIds = await this.serviceSettingsService.getServiceIdsByTemplateId(
+        args.where.project.id,
+        serviceTemplateId
+      );
+    }
+
     return this.prisma.resource.findMany({
       ...args,
       where: {
-        ...args.where,
+        ...where,
```

Comment:
you don't need the `delete` in line 1556  if you are using ...where here (and not using args.where)